src/app-data/appState.tsx

```diff
@@ -100,22 +101,22 @@ function actionHandler(draft: Draft<AppState>, action: AppStateAction) {
       return;
     case 'ADD_MEASUREMENTS': {
       mergeMeasurements(draft.data.measurements, action.payload);
-      for (let kind of Object.keys(action.payload) as MeasurementKind[]) {
-        if (!draft.view.selectedMeasurements[kind]) {
-          const measurement = getFirstMeasurementOrFail(
+
+      for (const kind of Object.keys(kindsLabel).filter(
+        (k) => k in action.payload,
+      ) as MeasurementKind[]) {
```

Comment:
What's the problem with `Object.keys(action.payload)` ?

Reply:
it's not ordered, in Measurement panel we use the kinds order of kindsLabel

Reply:
when we add measurements if no one is selected we have to select the first one in the first kind

Reply:
OK, thank you for the explanation!